File: C/linear search.c
#include <stdio.h>
#include <stdlib.h>
#define MAX 10
int main(){
    printf("Linear search");
    int a[MAX],b,c,key,i,n;
    do
    {
        printf("\n Enter the number of elements :");
        scanf("%d",&n);
        if (n>= MAX){
            printf("THE MAXIMUM SIZE IS %d",MAX);
            break;
        }
        for (i=0;i<n;i++){
            printf("\n Enter the element %d : ",i);
            scanf("%d",&b);
            a[i]=b;
        }
        printf("\n The elements stored in the array are :");
        for (i=0;i<n;i++){
            printf(" %d",a[i]);
        }
        printf("\n Enter the element to be searched :");
        scanf("%d",&key);
        for (i=0;i<n;i++){
            if (a[i]==key)
            {
                printf("\n The element is found at positon %d ",i);
                break;
            }
            else if(i==n-1 && a[i]!= key){
                 printf("\n The element is not found at any positon");
            }
        }
        printf("\n Do you want to continue again?");
        printf("\n If yes input 1 : ");
        scanf("%d",&c);
    } while (c==1);
    
    return 0;
}

File: JAVA/LinearSearch.java
public class LinearSearch{
    public static int search(int [] arr, int target){
        for (int i=0; i<arr.length;i++){
            if (arr[i]==target){
                return i;
            }
        }
        return -1;
    }
    public static void main(String args[]){
        int[] arr = {1,24,654,234,233,12,5};
        int target = 13;
        int res = search(arr, target);
        if (res == -1){
            System.out.println("Element Not found");
        }
        else{
            System.out.println("Element found at index "+res);
        }
    }
}